lib/sdf-server/src/service/v2/view.rs

```diff
@@ -81,7 +81,12 @@ impl IntoResponse for ViewError {
     fn into_response(self) -> Response {
         let (status_code, error_message) = match self {
             ViewError::NameAlreadyInUse(_) => (StatusCode::UNPROCESSABLE_ENTITY, self.to_string()),
-
+            ViewError::DalDiagram(
+                dal::diagram::DiagramError::DeletingLastGeometryForComponent(_, _),
+            )
+            | ViewError::Component(ComponentError::ComponentAlreadyInView(_, _)) => {
+                (StatusCode::FORBIDDEN, self.to_string())
```

Comment:
Is FORBIDDEN the right status code here? It feels like this is a input error

Reply:
@zacharyhamm 400? or 422?

Reply:
Axum uses 422 when you send it the wrong JSON, so i'd rather not overload that one

Reply:
I went by the wikipedia definition of it:

> [403 Forbidden](https://en.wikipedia.org/wiki/HTTP_403)
The request contained valid data and was understood by the server, but the server is refusing action. This may be due to the user not having the necessary permissions for a resource or needing an account of some sort, or **attempting a prohibited action (e.g. creating a duplicate record where only one is allowed)**. This code is also typically used if the request provided authentication by answering the WWW-Authenticate header field challenge, but the server did not accept that authentication. The request should not be repeated.

Reply:
If we want to keep 403 only for authorization levels or something like that we could change it though, but it feels like it makes sense on this case where the actions are forbidden by the application logic

Reply:
401 is authz